BatchApex/BatchApex.cls

```diff
@@ -0,0 +1,134 @@
+/**
+ * Collection of Closed Won Opportunities 
+ * The Primary contact associated with an account needs to know which all are the 
+ * closed won opportunities for his account per day. 
+ * Create a process which will run everyday in the midnight & collect all the 
+ * Closed won opportunities of that account in a day & send an email to Primary Contact. 
+ * The email body should contain the Opportunity Name & it's respective amount 
+ * with the aggregated amount in the end. 
+ */
+global with sharing class BatchApex implements Database.Batchable<SObject>, Database.Stateful, 
+    Schedulable {
+    Map<String, WrapperClass> accountOpportunityMap;
+    public static Double totalOpportunityAmount;
+    
+    /**
+     * Constructor to initialize the map to store opportunities and their sum
+     */
+    public BatchApex() {
+        accountOpportunityMap = new Map<String, WrapperClass>();
+    }
+    
+    /**
+     * Get all opportunities whos lookup field to account is not blank 
+     * and email field on Account object is not blank
+     */
+    global Database.QueryLocator start(Database.BatchableContext batchableContext) {
+            return Database.getQueryLocator([
+                SELECT 
+                    Id, 
+                    Name, 
+                    Amount, 
+                    StageName, 
+                    Opportunity.Account.Primary_Email__c, 
+                    AccountId 
+                FROM 
+                    Opportunity 
+                WHERE 
+                    AccountId <> null 
+                AND 
+                    Opportunity.Account.Primary_Email__c <> null
+                AND
+                    StageName = 'Closed Won'
+                AND
+                    CloseDate = TODAY
+            ]);
+    }
+    
+    /**
+     * This will sum up all the closed won opportunities of a perticular account per day
+     * and will send a mail to the primary contact
+     */
+    global void execute(
+        Database.BatchableContext batchableContext,
+        List<Opportunity> opportunityList) {
+        for (Opportunity opportunityInstance : opportunityList) {
+            if (accountOpportunityMap.containsKey(opportunityInstance.Account.Primary_Email__c)) {
```

Comment:
Why have you taken email? What will happen if we have 2 opportunities to Account 1 with email "abc.com" and 3 opportunities to Account 2 with email "abc"? Then in this case I think your code will make total of all 5 opportunities Amount correcsponding to both Accounts as their email Id is same. Which is wrong. We need some unique identifier for this Key. think over it and let me know. Also in this scenario the mail will be sent to which primary contact> Accont 1's or Account 2's. If you didnt understood my scenario or question then please ask me before implementing. I dont want u to waste ur time . First let me know the ans and then proceed